scipy/signal/_spectral_py.py

```diff
@@ -203,16 +207,16 @@ def lombscargle(
 
     # if no weights are provided, assume all data points are equally important
     if weights is None:
-        weights = np.ones_like(y, dtype=np.float64)
+        weights = np.ones_like(y, dtype=np.float64).squeeze()
```

Comment:
Why do we need to squeeze the inputs?

Reply:
The inputs must be 1D, which is the reason for the `x.ndim == 1` (along with the shapes matching) and `freqs.ndim == 1` checks. However, I think it made sense to just squeeze out any extraneous dimensions as a precaution, because the user would consider a shape(1, x) vector to be 1D—even though `np.ndim > 1`. I test for this in `test_valid_shape`.

Reply:
I understand but would prefer not to squeeze inputs. It can make it difficult to extend the functionality in the future

Reply:
No problem. Removed.